stable/mlrun/templates/_helpers.tpl

```diff
@@ -0,0 +1,87 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "mlrun.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "mlrun.fullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "mlrun.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "mlrun.labels" -}}
+helm.sh/chart: {{ include "mlrun.chart" . }}
+{{ include "mlrun.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end -}}
+
+{{/*
+Selector labels
+*/}}
+{{- define "mlrun.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "mlrun.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "mlrun.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+    {{ default (include "mlrun.fullname" .) .Values.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{- define "mlrun.apiName" -}}
+{{- printf "%s-api" (include "mlrun.fullname" .) | trunc 63 -}}
+{{- end -}}
+
+{{/*
+API Common labels
+*/}}
+{{- define "mlrun.apiLabels" -}}
+helm.sh/chart: {{ include "mlrun.chart" . }}
+{{ include "mlrun.apiSelectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end -}}
+
+{{/*
+API Selector labels
+*/}}
+{{- define "mlrun.apiSelectorLabels" -}}
+app.kubernetes.io/name: {{ include "mlrun.apiName" . }}
```

Comment:
app-name  (formerly app) label - is common to all mlrun and should be mlrun / chart name
I think you're missing (and should use in selectors) - `app.kubernetes.io/component` this will be `api` and unique to this sub component in this release